app/src/main/java/uk/nhs/nhsx/covid19/android/app/questionnaire/selection/QuestionnaireViewModel.kt

```diff
@@ -81,6 +81,25 @@ class QuestionnaireViewModel @Inject constructor(
         }
     }
 
+    fun onButtonCancelClicked(noSymptomsCallback: () -> Unit) {
+        val questions = getQuestions() ?: return
+        if (questions.any { it.isChecked }) {
+            AlertDialog.Builder(this)
```

Comment:
`AlertDialog.Builder(this)` doesn't look like it belongs in a ViewModel - I don't think this will compile?

Reply:
Alas github codespaces didn't provide me with that much insight, and it was probably too late in the evening for it. That's a good catch, and I don't think any of that belonged in the ViewModel. Moved it out, and just asking the model for what I need instead.